src/carp.c

```diff
@@ -808,6 +810,17 @@ int docarp(void)
     }
     strncpy(iface.ifr_name, interface, sizeof iface.ifr_name);
 #endif    
+    if (pidfile != NULL) {
+	pid_t pid = getpid();
+	FILE *_pidfile = fopen(pidfile, "w");
+	if (_pidfile != NULL) {
+	    char *buf = malloc(100 * sizeof (long));
```

Comment:
'bit short on time, hence briefly; I'd go along this:

```fwrite(&pid, sizeof(long), 1, pidfile);```

POSIX(2008) specifies that `pid_t` may not exceed `long`. One might test for the actual size of pid, though, as it's dependent on the OS installation/site admin (`PID_MAX` on Linuxen).